firewall/firewall.go

```diff
@@ -135,7 +137,10 @@ func handlePacket(pkt packet.Packet) {
 	// }
 
 	// allow local dns
-	if (pkt.Info().DstPort == 53 || pkt.Info().SrcPort == 53) && pkt.Info().Src.Equal(pkt.Info().Dst) {
+	if (pkt.Info().DstPort == 53 || pkt.Info().SrcPort == 53) &&
+		(pkt.Info().Src.Equal(pkt.Info().Dst) || // Windows redirects back to same interface
+			pkt.Info().Src.Equal(localhost4) || // Linux sometimes does 127.0.0.1->127.0.0.53
+			pkt.Info().Dst.Equal(localhost4)) {
```

Comment:
This is meant to match both directions. When both 127.0.0.1 and 127.0.0.x are used, we'll have to match both Src and Dst.

Reply:
Yeah but Source can only (normally) be 127.0.0.1 because that's the address assigned to the loopback interface. Destination may be something in 127.0.0.0/8 but if src **is 127.0.0.1** destination **MUST be in 127.0.0.0/8** there.